### ISSUE TITLE
Replace "Quit RetroArch: Quit the program." with "Quit RetroArch: Save modified settings and quit the program."

### ISSUE THREAD

**Issue by davidhedlund** [libretro/RetroArch#16000]
## Description

* In GNU/Linux the settings are closed when you close retorarch either from "Main Menu -> Quit RetroArch" or if you use the top right close window button
* In Android the settings in RA is not saved if you swipe-close the program. You have to quit retroarch from "Main Menu -> Quit RetroArch" to save the settings. But this is easy to forget.

### Expected behavior

It would be useful if  "Quit RetroArch: Quit the program." was replaced with "Quit RetroArch: Save the settings and quit the program."

### Actual behavior

![image](https://github.com/libretro/RetroArch/assets/6416073/23ced4ca-1f62-43e8-9bf6-8d4e286f295a)

### Steps to reproduce the bug
* Open RA
* In Home, scroll down to the "Quit RetroArch" entry

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
1.9.12

### Environment information

- OS: Android 10

**Comment by sonninnos** [comment]
Except it does not save if you disable the option that saves on quit, so in that case it would need 2 sublabels..

**Comment by hizzlekizzle** [comment]
F-Droid should have an up-to-date version.

**Comment by davidhedlund** [comment]
> > Except it does not save if you disable the option that saves on quit, so in that case it would need 2 sublabels..
> 
> Thank you. So to make it clear, the bold text below is desired under the specific circumstances:
> 
> If: Settings -> Configuration -> Save Configuration on Quit: Enabled [default]:
> 
>     * "Quit RetroArch: **Save the settings and** Quit the program."
> 
> 
> If: Settings -> Configuration -> Save Configuration on Quit: Disabled:
> 
>     * "Quit RetroArch: Quit the program **(Save Configuration on Quit disabled)**."
> 
> 
> > Also that 1.9.12 version is very much obsolete, but sure this thing still applies.
> 
> Oh. Well, this is the current version on both Google Play, and F-Droid.

This issue still persists, and this is a solution to it.

> F-Droid should have an up-to-date version.

Forget what I just said, I used an old phone.

**Comment by davidhedlund** [comment]
> F-Droid should have an up-to-date version.

You are right, the F-Droid release was 1.16.0. But the problem of this issue still applies.

**Comment by hizzlekizzle** [comment]
I think the main issue with this suggestion is that whether or not the settings are saved is contingent on the 'save settings on exit' setting, and I don't think we can make conditional menu text. That is, I believe it's hardcoded at compile time.

**Comment by davidhedlund** [comment]
> I don't think we can make conditional menu text.

Can you please try to verify if it's possible or not?

> That is, I believe it's hardcoded at compile time.

Sorry, I don't understand this, why not just modify the text before it's compiled?

**Comment by davidhedlund** [comment]
> What I mean is: that statement wouldn't always be true, depending on whether or not 'save on exit' is enabled or not,

Oh, I get your point: If there's nothing to save then it's pointless with the extra added "Save the settings" text to the menu entry. Well this could be solved by adding an if statement (display "Save the settings" if any setting has been modified).

> and we have to set the menu text at compile time rather than being able to change it dynamically depending on what settings are enabled.

Is it possible to change that?

**Comment by hizzlekizzle** [comment]
No, I mean this setting:
![image](https://github.com/libretro/RetroArch/assets/2805586/b5d9fc7e-64c3-4a31-ab6a-5046a2a71bda)
when it's OFF, "save the settings and quit the program" would not be true.

>is it possible to change that?
Almost anything is possible, it's more a matter of: how much work it would be to do it, what benefit comes from doing it and who has the inclination to do it?

**Comment by davidhedlund** [comment]
> ## Description
> 
>     * In GNU/Linux the settings are closed when you close retorarch even if you use the top right close window button
> 
>     * In Android the settings in RA is not saved if you swipe-close the program. You have to quit retroarch from "Home -> Quit RetroArch" to save the settings. But this is easy to forget.

A simple first step to solve this issue (_even_ if there's nothing to save) in Android, to be clear that you have to quit RA this way to save the settings.: Add
* "Quit RetroArch: Save the settings and quit the program"
* "Quit RetroArch: Save changed settings and quit the program" -- this will always be true, if nothing is changed nothing will be saved.

**Comment by davidhedlund** [comment]
> > Except it does not save if you disable the option that saves on quit, so in that case it would need 2 sublabels..
> 
> Thank you. So to make it clear, the bold text below is desired under the specific circumstances:
> 
> If: Settings -> Configuration -> Save Configuration on Quit: Enabled [default]:
> 
>     * "Quit RetroArch: **Save the settings and** Quit the program."
> 
> 
> If: Settings -> Configuration -> Save Configuration on Quit: Disabled:
> 
>     * "Quit RetroArch: Quit the program **(Save Configuration on Quit disabled)**."
> 
> 
> > Also that 1.9.12 version is very much obsolete, but sure this thing still applies.
> 
> Oh. Well, this is the current version on both Google Play, and ~F-Droid~ [incorrect].

@hizzlekizzle I discussed that here^, but yes that requires dynamic code.

**Comment by davidhedlund** [comment]
> No, I mean this setting: ![image](https://private-user-images.githubusercontent.com/2805586/295081738-b5d9fc7e-64c3-4a31-ab6a-5046a2a71bda.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3NjY3NzcsIm5iZiI6MTcwNDc2NjQ3NywicGF0aCI6Ii8yODA1NTg2LzI5NTA4MTczOC1iNWQ5ZmM3ZS02NGMzLTRhMzEtYWI2YS01MDQ2YTJhNzFiZGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMDIxNDM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTU1NTEwMjdhMTNhMmI3Yjg1YjlhMGYzZjIyMWVlYjQxZDI2ZDAzNjMzZDViN2M5MWJlN2QyNmUzMGEzZTU3OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.t56SRo2S8A9dglMGkNq-nhMz7nKYic8OehmU-X2AGiE) when it's OFF, "save the settings and quit the program" would not be true.
> 
> > is it possible to change that?
> 
> Almost anything is possible, it's more a matter of: how much work it would be to do it, what benefit comes from doing it and who has the inclination to do it?

I discussed that [here](https://github.com/libretro/RetroArch/issues/16000#issuecomment-1852267869)

**Comment by davidhedlund** [comment]
* https://github.com/libretro/RetroArch/issues/16094

**Comment by RobLoach** [comment]
I don't believe we have to be so verbose here. If anything, I think we could remove the "RetroArch" from the menu label, as it's implied and explained in the sublabel.

![quit-app](https://github.com/libretro/RetroArch/assets/25086/0673a5da-a005-4ac2-aeb5-29d6f4275a74)

**Comment by davidhedlund** [comment]
> I don't believe we have to be so verbose here. It does a few things when exiting, which is configurable, and I don't think we have to explain that. It's unnecessary bloat... If anything, I think we could remove the "RetroArch" from the menu label, as it's implied and explained in the sublabel.
> 
> ![quit-app](https://private-user-images.githubusercontent.com/25086/296608484-0673a5da-a005-4ac2-aeb5-29d6f4275a74.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyNTY4MjAsIm5iZiI6MTcwNTI1NjUyMCwicGF0aCI6Ii8yNTA4Ni8yOTY2MDg0ODQtMDY3M2E1ZGEtYTAwNS00YWMyLWFlYjUtMjlkNmY0Mjc1YTc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDE4MjIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmZGJlNGY5M2JhZWY5OWYyNGRiZmRjODUwYjMzYjU5MmFiMzc4OGI4NGYwMjc0ZWQzM2ViODhjNzI0ZjZiMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.y6rnQ3bgRGI0YagDIpGiLpIHG57v01oKM3k7zRSICdk)

I respect your opinion. However, this issue is quite detailed. So, to avoid confusion, do you mind moving your idea (I like it) to a new issue?

**Comment by davidhedlund** [comment]
* https://github.com/libretro/RetroArch/issues/16119

**Comment by hizzlekizzle** [comment]
I think the verbosity is mostly to intended to disambiguate from the 'restart' and 'close content' entries in the quick menu. Nevertheless, I think there's a reasonable argument that it's unnecessarily verbose, though that has largely been our strategy for improving usability for n00bs: explain everything.

**Comment by davidhedlund** [comment]
From https://github.com/libretro/RetroArch/pull/16144#issuecomment-1902834630

The current text "`Configuration save on exit is enabled/disabled.`" does not specify which configuration that is enabled. It should be updated to
* `Save Configuration on Quit is [enabled]/[disabled]` or
* `Save Configuration on Quit is  [enabled]/[disabled], Save Remap Files on Quit is  [enabled]/[disabled]`

**Comment by davidhedlund** [comment]
> If: Settings -> Configuration -> Save Configuration on Quit: Enabled [default]:
> 
>     * Label: Quit. Sub label: "Quit RetroArch application and save settings."
> 
> 
> If: Settings -> Configuration -> Save Configuration on Quit: Disabled:
> 
>     * Label: "Quit. Sub label: Save Configuration on Quit is disabled."


@zoltanvb [almost solve this](https://github.com/libretro/RetroArch/pull/16144), which is great, but the current text that added: "`_Configuration save on exit_ is enabled/disabled.`" does not specify which configuration that is enabled. It should be updated to
* `Save Configuration on Quit is [enabled]/[disabled]` or
* `Save Configuration on Quit is  [enabled]/[disabled], Save Remap Files on Quit is  [enabled]/[disabled]`

**Comment by davidhedlund** [comment]
* https://github.com/libretro/RetroArch/issues/16349

**Comment by davidhedlund** [comment]
Fixed. The sub-label

> Quit RetroArch.

has been replaced with:
> Quit RetroArch. Configuration save on exit is [enabled]/[disabled].
